app/routes.py

```diff
@@ -1 +1,121 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint
+from flask import Blueprint, jsonify, request, abort, make_response
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal 
+from datetime import datetime
+import requests
+import os
+
+tasks_bp = Blueprint("tasks", __name__,url_prefix="/tasks")
+
+# Helper function
+def get_model_from_id(cls, model_id):
+    try:
+        model_id = int(model_id)
+    except ValueError:
+        return abort(make_response({"msg": f"invalid data type: {model_id}"}, 200))
+
+
+    chosen_task = cls.query.get(model_id)
+
+    if chosen_task is None:
+        return abort(make_response({"msg": f"Could not find the task with id: {model_id}"}, 404))
+    
+    return chosen_task
+
+@tasks_bp.route('', methods=['POST'])
+def create_one_task():
+    request_body = request.get_json()
+    if "title" not in request_body or "description" not in request_body:
+        return abort(make_response({"details": "Invalid data"}, 400))
+
+    new_task= Task(title=request_body["title"],
+                description=request_body["description"]
+
+    )
+
+    db.session.add(new_task)
+    db.session.commit()
+
+    return jsonify({"task":new_task.to_dict()}), 201
+
+@tasks_bp.route('', methods=['GET'])
+def get_all_tasks():
+
+    sort_query_value = request.args.get('sort')
+    if sort_query_value == "desc":
+        tasks = Task.query.order_by(Task.title.desc()).all() 
+    else:
+    
+        tasks = Task.query.order_by(Task.title.asc()).all()
+
+    result = []
+
+    for task in tasks:
+        result.append(task.to_dict())
+    
+    return jsonify(result), 200
+
+@tasks_bp.route('/<task_id>', methods=['GET'])
+def get_one_task(task_id):
+
+    chosen_task = get_model_from_id(Task, task_id)
+
+    return jsonify({"task":chosen_task.to_dict()}), 200
+
+@tasks_bp.route('/<task_id>', methods=['PUT'])
+def update_one_task(task_id):
+    update_task = get_model_from_id(Task, task_id)
+
+    request_body = request.get_json()
+
+    try:
+        update_task.title = request_body["title"]
+        update_task.description = request_body["description"]
+
+    except KeyError:
+        return jsonify ({"msg": f"Missing attributes"}), 400
```

Comment:
Good use of try-except, but it would be very good to indicate to the user what attribute was wrong/missing.